### ISSUE TITLE
code for sinusoid_encoding_table

### ISSUE THREAD

**Issue by hqh1997** [MatteoStefanini/DNAPerceiver#1]
hi, your work is excellent, well done. but how to define the code for sinusoid_encoding_table https://github.com/MatteoStefanini/DNAPerceiver/blob/b7047584e91c7cf4259e54526d7a04a6f860bc7a/models/perceiver.py#L9C48-L9C48. I tried to realize it, is this right?
ef sinusoid_encoding_table(n_position, dim, dtype=torch.float32):
    """
    Generate sinusoidal encoding table.

    Args:
        n_position (int): The maximum number of positions to be encoded.
        dim (int): The dimension of the encoding.
        dtype: Data type of the encoding.

    Returns:
        torch.Tensor: Sinusoidal encoding table.
    """
    position = torch.arange(0, n_position).unsqueeze(1).type(dtype)
    div_term = torch.exp(torch.arange(0, dim, 2).type(dtype) * -(math.log(10000.0) / dim))
    enc_table = torch.zeros(n_position, dim).type(dtype)
    enc_table[:, 0::2] = torch.sin(position * div_term)
    enc_table[:, 1::2] = torch.cos(position * div_term)
    return enc_table
   by the way, can you share some case data for us to run your code.Looking forward to your reply
